site/docs/architecture/index.md

```diff
@@ -0,0 +1,35 @@
+---
+sidebar_position: 2
+---
+
+# Architecture
+
+**Gerust is built with API servers in mind.** We expect it to be used in projects that provide an HTTP interface (likely JSON, e.g. REST, although Gerust makes not assumption on the exact data format used) over a network. The API server might provide access to data in a database but Gerust can also be used for projects with a database, e.g. as a proxy in front of other services. We specifically **did not create Gerust for website projects** that render HTML directly, or handle e.g. form submissions originating from browsers directly.
```

Comment:
To keep everything in a single stack?

That said, I do like a lean framework that does a single thing well...

Reply:
I can't see a real-life example where the benefits of Rust that justify the investment (at this point almost none of the web/backend teams have Rust experience so it's an investment for sure) on the backend side (stability, performance, maintainability) are equally important on the UI side or for something that renders HTML. If you're building an actual web app with a minimum level of functionality and behavior in the browser I can't see how you would build that with Rust at all or how that would not be dramatically worse and harder than e.g. SvelteKit, htmx or whatever.

That said, we can still support that in Gerust at a later point but for now, we need to narrow the focus down anyway to have a realistic chance to build sth. that's a real option for a particular kind of project.

Reply:
JS-free or htmx-based websites have their use. Rather than putting people off that want to serve HTML or do forms or something, we can state that Gerust is optimized for REST-APIs or something instead.

Reply:
added a note saying we might change in the future